localstack/dev/kubernetes/__main__.py

```diff
@@ -0,0 +1,198 @@
+import os
+
+import click
+import yaml
+
+from localstack import version as localstack_version
+
+
+def generate_k8s_cluster_config(pro: bool = False, mount_moto: bool = False):
+    volumes = []
+    root_path = os.path.join(os.path.dirname(__file__), "..", "..", "..")
+    localstack_code_path = os.path.join(root_path, "localstack")
+    volumes.append(
+        {
+            "volume": f"{os.path.normpath(localstack_code_path)}:/code/localstack",
+            "nodeFilters": ["server:*", "agent:*"],
+        }
+    )
+
+    egg_path = os.path.join(root_path, "localstack_core.egg-info/entry_points.txt")
+    if pro:
+        ext_path = os.path.join(root_path, "..", "localstack-ext")
+        ext_code_path = os.path.join(ext_path, "localstack_ext")
+        egg_path = os.path.join(ext_path, "localstack_ext.egg-info/entry_points.txt")
+
+        volumes.append(
+            {
+                "volume": f"{os.path.normpath(ext_code_path)}:/code/localstack_ext",
+                "nodeFilters": ["server:*", "agent:*"],
+            }
+        )
+
+    volumes.append(
+        {
+            "volume": f"{os.path.normpath(egg_path)}:/code/entry_points",
+            "nodeFilters": ["server:*", "agent:*"],
+        }
+    )
+
+    if mount_moto:
+        moto_path = os.path.join(root_path, "..", "moto", "moto")
+        volumes.append(
+            {"volume": f"{moto_path}:/code/moto", "nodeFilters": ["server:*", "agent:*"]}
+        )
+
+    config = {"apiVersion": "k3d.io/v1alpha3", "kind": "Simple", "volumes": volumes}
+
+    return config
+
+
+def snake_to_kebab_case(string: str):
+    return string.lower().replace("_", "-")
+
+
+def generate_k8s_cluster_overrides(
+    pro: bool = False, cluster_config: dict = None, write: bool = False
+):
+    volumes = []
+    for volume in cluster_config["volumes"]:
+        name = snake_to_kebab_case(volume["volume"].split(":")[-1].split("/")[-1])
+        volume_type = "Directory" if name != "entry-points" else "File"
+        volumes.append(
+            {
+                "name": name,
+                "hostPath": {"path": volume["volume"].split(":")[-1]},
+                "type": volume_type,
+            }
+        )
+
+    volume_mounts = []
+    target_path = "/opt/code/localstack/"
+    venv_path = os.path.join(target_path, ".venv", "lib", "python3.11", "site-packages")
+    for volume in volumes:
+        if volume["name"] == "entry-points":
+            project = "localstack_ext-" if pro else "localstack_core-"
+            version = localstack_version.__version__
+            dist_info = f"{project}{version}0.dist-info"
+            volume_mounts.append(
+                {
+                    "name": volume["name"],
+                    "readOnly": True,
+                    "mountPath": os.path.join(venv_path, dist_info, "entry_points.txt"),
```

Comment:
Annoyingly this is not the path of the community `entry_points.txt` path. It is at `/opt/code/localstack/localstack_core.egg-info/entry_points.txt`.